tutorials/single-page-tutorials/nested-nfts-with-wallet-lib.md

```diff
@@ -0,0 +1,423 @@
+  try {
+    // Read file from path
+    const file = fs.readFileSync(filePath);
+
+    // Upload file to IPFS
+    const fileAdded = await ipfsNode.add(file);
+    const contentIdentifier = fileAdded.path;
+
+    console.log('\n');
+    console.log(consoleColor, `Your file was successfully uploaded to IPFS:`);
+    console.log(`https://ipfs.io/ipfs/${contentIdentifier}`, '\n');
+
+    return contentIdentifier;
+  } catch (error) {
+    console.error('IPFS upload error', error);
+  }
+}
+```
+
+#### 4. Mint NFT
+
+The function `mintNFT()` takes a NFT name, the previously created and unique IPFS identifier and an unlocked Stronghold account in order to mint an NFT on the configurated network. The metadata will be stored in the immutable metadata field, meaning it will remain unchanged for as long as the NFT itself exists on the network. Also note that the function does not pass a specific address to the Stronghold Account method `mintNfts()`. By default the method will mint all NFTs on the first address of the used Stronghold account, as long as it holds enough funds for Storage Deposit.
+
+:::note
+
+The NFT metadata follows [Tangle Improvement Proposal (TIP 27)](https://github.com/Kami-Labs/tips/blob/main/tips/TIP-0027/tip-0027.md).
+
+:::
+
+```javascript
+// This function creates NFT metadata and mints a new NFT
+async function mintNFT(itemName, ipfsCid, StrongholdAccount) {
+  // Define NFT metadata
+  const metadataObject = {
+    standard: 'IRC27',
+    type: 'image/jpeg',
+    version: 'v1.0',
+    name: `${itemName}`,
+    uri: `https://ipfs.io/ipfs/${ipfsCid}`,
+  };
+
+  const metadataBytes = utf8ToHex(JSON.stringify(metadataObject));
+
+  const response = await StrongholdAccount.mintNfts([
+    {
+      immutableMetadata: metadataBytes,
+    },
+  ]);
+
+  console.log(consoleColor, `Your NFT was successfully minted in this block:`);
+  console.log(`${explorerURL}/block/${response.blockId}`, '\n');
+}
+```
+
+#### 5. Main Script
+
+This is the main function, which loads and unlocks a Stronghold account, triggers the IPFS upload and mints a NFT for every file provided in a list. In this case we will mint three NFTs (avatar, mask, weapon).
+
+```javascript
+// This function loads a Stronghold account and triggers the IPFS upload as well as NFT minting process for a list of files
+async function main() {
+  try {
+    const manager = new AccountManager({
+      storagePath: `./${accountName}-database`,
+    });
+    await manager.setStrongholdPassword(password);
+    const StrongholdAccount = await manager.getAccount(accountName);
+    await StrongholdAccount.sync();
+
+    const ipfsNode = await startIpfsNode();
+
+    const items = ['avatar', 'mask', 'weapon'];
+
+    // This loop through the list of file names
+    for (const item of items) {
+      let cid = await uploadToIPFS(ipfsNode, `${item}.jpg`);
+      await mintNFT(item, cid, StrongholdAccount);
+    }
+  } catch (error) {
+    console.log('Error: ', error);
+  }
```

Comment:
Same as above, if we are just logging the error, I'd just let the default error handling do its thing and remove the try/catch block.